_aicspylibczi/TypedImage.h

```diff
@@ -126,7 +126,7 @@ TypedImage<T>::loadImage(const std::shared_ptr<libCZI::IBitmapData>& bitmap_ptr_
   // height, they are both uint32_t and you'll get an overflow for larger images
   uint8_t* sEnd = static_cast<uint8_t*>(lckScoped.ptrDataRoi) + lckScoped.size;
   if ((lckScoped.stride % size_.w == 0) && ( m_pixelType == libCZI::PixelType::Gray8 || m_pixelType == libCZI::PixelType::Gray16 ||
-      m_pixelType == libCZI::PixelType::Gray32Float)) {
+      m_pixelType == libCZI::PixelType::Gray32Float) && channels_ != 3 ) {
```

Comment:
Slightly confused - is `channels_` the number of image channels?  what happens if you have a 3 channel image?

Reply:
channels_ in this case is the number of channels in the subblock. Since a subblock is 1 or 3 channels (gray8 vs BGR24) the larger concept of channels isn't applicable here.

Reply:
I changed the variable to be `samples_per_pixel_` for better clarity

Reply:
I see, thanks!